polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala

```diff
@@ -198,10 +200,23 @@ class SocketTransport(
   forceServerAddress: Option[String] = None
 ) extends Transport[InetSocketAddress] {
 
-  private def openServerChannel: RIO[Blocking, ServerSocketChannel] = effectBlocking {
-    ServerSocketChannel.open().bind(
-      new InetSocketAddress(
-        forceServerAddress.getOrElse(java.net.InetAddress.getLocalHost.getHostAddress), 0))
+  private[this] def socketIsFree(host: String, port: Int): Boolean =
+    try {
+      new Socket(host, port).close()
+      false
+    } catch {
+      case _: ConnectException =>
+        true
+    }
+
+  private def openServerChannel(
+                                 config: PolynoteConfig
+                               ): RIO[Blocking, ServerSocketChannel] = effectBlocking {
+    val address = config.kernel.listen.getOrElse(java.net.InetAddress.getLocalHost.getHostAddress)
```

Comment:
Looks like we lost `forceServerAddress` here... maybe we should just remove that parameter?

Reply:
Yes that's true. Do you mean from the configuration?

Reply:
I mean the parameter `forceServerAddress` isn't really needed anymore, since it can be specified in configuration.